src/sinks/util/uri.rs

```diff
@@ -0,0 +1,62 @@
+use http::Uri;
+use serde::{
+    de::{Error, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use std::fmt;
+
+/// A wrapper for `http::Uri` that implements the serde traits.
+#[derive(Default, Debug, Clone)]
+pub struct UriSerde(Uri);
+
+impl Serialize for UriSerde {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let uri = format!("{}", self.0);
+        serializer.serialize_str(&uri)
+    }
+}
+
+impl<'a> Deserialize<'a> for UriSerde {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'a>,
+    {
+        deserializer.deserialize_string(UriVisitor)
+    }
+}
+
+impl fmt::Display for UriSerde {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+struct UriVisitor;
+
+impl<'a> Visitor<'a> for UriVisitor {
+    type Value = UriSerde;
+
+    fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "a string containing a HTTP")
```

Comment:
Typo?

Reply:
HAHAH wow I guess my editor didn't save the rest of my sentence.